### PR TITLE
Tooltip when hovering over atlas images

### LINKED ISSUE #26: Allow user to hover over a brain region and see the metadata.
This is currently implemented within brainreg-segment:

https://github.com/brainglobe/brainreg-segment/blob/a84d4b2fab53c256590b79d57beb1eac5810ef0f/brainreg_segment/segment.py#L91-L105

Ideally this would:
1. use tooltips instead (https://github.com/brainglobe/brainreg-segment/issues/89)
2. Be visible regardless of what layers are added/visible (as long as one is!), not just the annotations layer

I think for 1. the metadata would have to be added to the annotation layer itself

### PR DESCRIPTION
**Pull request by alessandrofelder**
## Description

**What is this PR**

- [ ] Bug fix
- [x] Addition of a new feature
- [ ] Other

**Why is this PR needed?**
Being able to display the region info when hovering over any atlas image in 2d is a sought-after feature.

**What does this PR do?**
(This PR is a workaround until napari allows[ developers to provide custom tooltips for the selected layer](https://forum.image.sc/t/napari-tooltip-for-cursor-when-moving-around-viewer/85142/12). We can track this sort of thing in meta-issue #87)

An initial implementation of a layer tooltip. 
* implemented as a QLabel that appears when moving the mouse over any atlas images (annotation, default reference, additional reference). 
* displayed text matches what `brainreg-segment` displays as help
* `napari_atlas_representation` constructor switches on all layer tooltips in napari

## References
Closes #26 

## How has this PR been tested?

Two additional tests added that cover added functionality
Existing tests pass
Manual testing looks reasonable

## Is this a breaking change?
No

## Does this PR require an update to the documentation?

Opened an issue in the docs https://github.com/brainglobe/website/issues/59

## Checklist:

- [x] The code has been tested locally
- [x] Tests have been added to cover all new functionality (unit & integration)
- [ ] The documentation has been updated to reflect any changes (see https://github.com/brainglobe/website/issues/59)
- [x] The code has been formatted with [pre-commit](https://pre-commit.com/)

### CODE REVIEW

**Comment by adamltyson** [comment]
@alessandrofelder this works for me, but for some reason the brain region tooltip is displayed with an offset. The other tooltips in the widget work fine. 
Screenshot from 2023-08-31 12-19-09

File: brainrender_napari/napari_atlas_representation.py [hunk 1]
```diff
@@ -15,23 +18,40 @@ class NapariAtlasRepresentation:
     mesh_opacity: float = 0.4
     mesh_blending: str = "translucent_no_depth"
 
+    def __post_init__(self) -> None:
+        """Setup a custom QLabel tooltip and enable napari layer tooltips"""
+        self._tooltip = QLabel(self.viewer.window.qt_viewer.parent())
```
Comment (adamltyson) [inline brainrender_napari/napari_atlas_representation.py:23]: This line produces a deprecation warning:
```
FutureWarning: Public access to Window.qt_viewer is deprecated and will be removed in
v0.5.0. It is considered an "implementation detail" of the napari
application, not part of the napari viewer model. If your use case
requires access to qt_viewer, please open an issue to discuss.!
```

Comment (alessandrofelder) [reply brainrender_napari/napari_atlas_representation.py:23]: Yes - I don't think we have a nice way around this if we want a tooltip until napari allows custom layer tooltips. It's something that should be easy to contribute to napari before 0.5.0 and then fix here, but better to get the tool to users earlier, I think?

**Comment by alessandrofelder** [comment]
> the brain region tooltip is displayed with an offset. The other tooltips in the widget work fine. 
Thanks, that offset shouldn't be as big! I'll try to fix!

File: brainrender_napari/napari_atlas_representation.py (hunk 1 above)
Comment (adamltyson) [reply brainrender_napari/napari_atlas_representation.py:23]: Agreed, but is there an ETA for v0.5.0? If it's soon, then we'll have an issue.

Comment (alessandrofelder) [reply brainrender_napari/napari_atlas_representation.py:23]: from a zulip convo in June, it's not soon.

**Review by adamltyson — APPROVED** [review]

**Pull request merged**